### DISCUSSION TITLE
How to use a napalm community driver with netbox

### DISCUSSION THREAD

**Discussion by da-me** [netbox-community/netbox#5807] [Q&A] [answered]
I'm trying to configure netbox to use napalm and a community driver (ruckus fastrion). I have installed both napalm and the ruckus community driver, and everything looks fine from napalms point of view:

> ~ $ sudo -u netbox napalm --user user--debug --vendor napalm_ruckus_fastiron 1.2.3.4 call get_facts
> Enter password:
> 2021-02-12 10:09:53,961 - napalm - DEBUG - Starting napalm's debugging tool
> 2021-02-12 10:09:53,961 - napalm - DEBUG - Gathering napalm packages
> 2021-02-12 10:09:53,962 - napalm - DEBUG - napalm-ros==1.0.1
> 2021-02-12 10:09:53,962 - napalm - DEBUG - napalm-ruckus-fastiron==1.0.26
> 2021-02-12 10:09:53,962 - napalm - DEBUG - napalm-vyos==0.2.1
> 2021-02-12 10:09:53,962 - napalm - DEBUG - napalm==3.2.0
> 2021-02-12 10:09:53,962 - napalm - DEBUG - get_network_driver - Calling with args: ('napalm_ruckus_fastiron',), {}
> 2021-02-12 10:09:53,965 - napalm - DEBUG - get_network_driver - Successful
> 2021-02-12 10:09:53,965 - napalm - DEBUG - __init__ - Calling with args: (<class 'napalm_ruckus_fastiron.FastIron.FastIronDriver'>, '1.2.3.4', 'user'), {'password': '*******', 'timeout': 60, 'optional_args': {}}
> 2021-02-12 10:09:53,965 - napalm - DEBUG - __init__ - Successful
> 2021-02-12 10:09:53,965 - napalm - DEBUG - pre_connection_tests - Calling with args: (<napalm_ruckus_fastiron.FastIron.FastIronDriver object at 0x7fc1d75bc4c0>,), {}
> 2021-02-12 10:09:53,965 - napalm - DEBUG - open - Calling with args: (<napalm_ruckus_fastiron.FastIron.FastIronDriver object at 0x7fc1d75bc4c0>,), {}
> SSH connection established to 1.2.3.4:22
> Interactive SSH session established

So the napalm command line tool works and talks to the switch, and trying python directly looks fine to:

> >>> import napalm
> >>> napalm.get_network_driver('napalm_ruckus_fastiron')
> <class 'napalm_ruckus_fastiron.FastIron.FastIronDriver'>

The problem with netbox is: when I try to use'napalm_ruckus_fastiron' as a driver name in a platform definition and check the system status of the device I get the following error:

> {"detail":"NAPALM driver for platform Ruckus Switch not found: napalm_ruckus_fastiron."}

Trying the api call directly gives the same result:

> GET /api/dcim/devices/190/napalm/?method=get_facts
> 
> HTTP 503 Service Unavailable
> Allow: GET, HEAD, OPTIONS
> Content-Type: application/json
> Vary: Accept
> 
> {
>     "detail": "NAPALM driver for platform Ruckus Switch not found: napalm_ruckus_fastiron."
> }
> 

I'm using netbox v2.10.3 on an Ubuntu 20.04 system, napalm 3.2.0 and napalm_ruckus_fastiron 1.0.26.

I tried every combination of napalm_ruckus_fastiron, ruckus_fastiron, fastiron and FastIronDriver I can think of.

What is the correct driver name for this community driver? Is there something I'm missing or some obvious mistake on my end, or is it something that needs to be adressed with the author of the ruckus fastiron community driver?

**Comment by candlerb** [comment, marked as answer]
Are you sure the napalm_ruckus_fastiron driver is installed inside your netbox virtualenv? 

Your test line suggests it is working *outside* the virtualenv:

```
~ $ sudo -u netbox napalm --user user--debug --vendor napalm_ruckus_fastiron 1.2.3.4 call get_facts
```

You can test like this:

```
sudo -u netbox bash
cd /opt/netbox
source venv/bin/activate
napalm --user user ... etc
```

The fix is to add the required package(s) to `local_requirements.txt` and run `./upgrade.sh`

**Reply by da-me** [reply]
Thank you! That was indeed my mistake, thanks for pointing me in in the right direction!
